src/mixins/soft-crud.repository.mixin.ts

```diff
@@ -1,342 +1,22 @@
-import {Constructor, Getter} from '@loopback/core';
-import {
-  AndClause,
-  Condition,
-  DataObject,
-  DefaultCrudRepository,
-  Entity,
-  Filter,
-  OrClause,
-  Where,
-} from '@loopback/repository';
-import {Count} from '@loopback/repository/src/common-types';
-import {HttpErrors} from '@loopback/rest';
-import {Options} from 'loopback-datasource-juggler';
+import {DefaultCrudRepository, Entity} from '@loopback/repository';
 
-import {ErrorKeys} from '../error-keys';
-import {SoftDeleteEntity} from '../models';
 import {
-  AbstractConstructor,
+  Constructor,
   IBaseEntity,
   ISoftCrudRepositoryMixin,
-  IUser,
+  MixinBaseClass,
 } from '../types';
+import {SoftCrudRepository} from '../repositories';
+import {extendPrototype} from '../decorators/soft-crud.decorator';
 
 export function SoftCrudRepositoryMixin<
   E extends Entity & IBaseEntity,
   ID,
-  T extends
-    | Constructor<DefaultCrudRepository<E, ID, R>>
-    | AbstractConstructor<DefaultCrudRepository<E, ID, R>>,
+  T extends MixinBaseClass<DefaultCrudRepository<E, ID, R>>,
```

Comment:
Shouldn't this be the last argument to stay consistent with the repo generic arguments

Reply:
True but doing so would break existing implementations as it was in the same sequence before as well.

Reply:
Oh, then we would have to release a breaking change for this as well in the future.

Reply:
Noted.

Reply:
Vese we can't do this because of the following ts rule:

> Required type parameters may not follow optional type parameters

And `Relations` type after this has the default value of `{}`

Reply:
Oh then ignore this.